### PR TITLE
Integrate with new POST request for AP with feature flag

### PR DESCRIPTION
**Pull request by mjoyce91**
- [x] Saved searches still generate notifications, with combination of saved searches from old GET and new POST
- [x] Favorites work
- [x] Similar positions work

Use with https://github.com/MetaPhase-Consulting/mock-fsbid/pull/199

### CODE REVIEW

**Comment by mjoyce91** [comment]
Azure is currently using this branch and integration looks good

File: talentmap_api/fsbid/services/available_positions.py [hunk 1]
```diff
@@ -318,66 +342,74 @@ def fsbid_ap_to_talentmap_ap(ap):
...
 
-def convert_ap_query(query, allowed_status_codes=["HS", "OP"], isTandem=False):
+def convert_ap_query(query, allowed_status_codes=["HS", "OP"], isTandem=False, use_post=False):
     '''
     Converts TalentMap filters into FSBid filters
     '''
+
+    prefix = ""
+    if not use_post:
+        prefix = "request_params."
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/available_positions.py:352]: 🦜

File: talentmap_api/fsbid/services/common.py [hunk 2]
```diff
@@ -187,6 +194,19 @@ def get_results(uri, query, query_mapping_function, jwt_token, mapping_function,
         return response.get("Data", {})
 
 
+def get_results_with_post(uri, query, query_mapping_function, jwt_token, mapping_function, api_root=API_ROOT):
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/common.py:197]: Should out `api_root` default to `CP_API_V2_URL` since we should only be hitting this for POST?

Comment (mjoyce91) [reply talentmap_api/fsbid/services/common.py:197]: I can actually see us using this endpoint elsewhere in the future (if other endpoints gets converted to POST), so would rather leave to be generic

Comment (elizabeth-jimenez) [reply talentmap_api/fsbid/services/common.py:199]: makes sense!

**Review by elizabeth-jimenez — APPROVED** [review]
🍬 👍

**Comment by elizabeth-jimenez** [comment]
@mjoyce91 Code looked good, but couldn't spin up cause of Mock image issue; Approved but will leave the merging decision to you 🦜 period.

File: talentmap_api/fsbid/services/common.py [hunk 3]
```diff
@@ -205,29 +225,33 @@ def get_fsbid_results(uri, jwt_token, mapping_function, email=None):
...
     return pydash.get(response, '[0]') or None
 
 
-def send_get_request(uri, query, query_mapping_function, jwt_token, mapping_function, count_function, base_url, host=None, api_root=API_ROOT):
+def send_get_request(uri, query, query_mapping_function, jwt_token, mapping_function, count_function, base_url, host=None, api_root=API_ROOT, use_post=False):
     '''
     Gets items from FSBid
     '''
     pagination = get_pagination(query, count_function(query, jwt_token)['count'], base_url, host) if count_function else {}
+    get = get_results_with_post if use_post else get_results
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/common.py:242]: Can we rename this `get`? 😅

File: talentmap_api/fsbid/services/common.py [hunk 4]
```diff
@@ -205,29 +225,33 @@ def get_fsbid_results(uri, jwt_token, mapping_function, email=None):
     return map(mapping_function, response.get("Data", {}))
 
 
-def get_individual(uri, id, query_mapping_function, jwt_token, mapping_function, api_root=API_ROOT):
+def get_individual(uri, id, query_mapping_function, jwt_token, mapping_function, api_root=API_ROOT, use_post=False):
     '''
     Gets an individual record by the provided ID
     '''
-    response = get_results(uri, {"id": id}, query_mapping_function, jwt_token, mapping_function, api_root)
+    get = get_results_with_post if use_post else get_results
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/common.py:232]: and this one?

**Pull request merged**